### DISCUSSION TITLE
Displaying current path when saving query

### DISCUSSION THREAD

**Discussion by bjornasm** [tconbeer/harlequin#511] [Ideas (Feature Requests)]
This might be more fitting for the textual repo, I am not sure - but I throw it out there regardless. When saving a query it feels a bit like throwing it out in the dark. First I typed out the whole path to where it should be saved (autocomplete would be nice here as well!), but then I wagered to just write the filename and press save. The file was saved to the directory where I started Harlequin from (which might be self evident for some), which was nice. However it would be nice to have some confirmation that this is the case, maybe a greyed out underlay text of the current path or just the current path already written in there for the user to fill out their filename or navigate elsewhere?

**Comment by tconbeer** [comment]
Thanks for the feedback! (This is the right repo for this)

I think showing the currently-computed file path is a good idea. We could probably re-use the space for error/validation messages for this (immediately above the input box).  Created https://github.com/tconbeer/textual-textarea/issues/232 to track this specifically.

re: autocomplete, you should get this once you start typing a unique path. So, for  example, from the root of this project, typing `./t` shows `./tests` and you can press <kbd>tab</kbd> to complete:
![image](https://github.com/tconbeer/harlequin/assets/705906/cd393fc3-f136-4f4a-bb51-a095c3a75af9)
typing `./s` does not suggest `./src`, since there are also directories called `static` and `stubs` to choose from.

**Reply by bjornasm** [reply]
Thanks, was not aware of the autocomplete! Good solution regarding the file path as well, can try to look at the code myself if you got a long list with things to do before that.

**Reply by tconbeer** [reply]
The change would be to this event handler:

https://github.com/tconbeer/textual-textarea/blob/a4c3291c77c48192f2f2c1aa2d8fbdc988d15bf3/src/textual_textarea/text_editor.py#L1132-L1139

**Reply by bjornasm** [reply]
Think this might be an ok implementation https://github.com/tconbeer/textual-textarea/pull/233
